homework-selects/sql_questions.sql

```diff
@@ -3,55 +3,460 @@
---the employees who gets the salary as the salary earn by the employee which is maximum within the joining person January 1st, 2002 and December 31st, 2003.  
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name, salary, department_name, city
+FROM employees
+    INNER JOIN departments USING (department_id)
+    INNER JOIN locations USING (location_id);
+
+--the employees who gets the salary as the salary earn by the employee which is maximum within the joining person January 1st, 2002 and December 31st, 2003.
+SELECT *
+FROM employees
+WHERE salary = (
+    SELECT MAX(salary)
+    FROM employees
+    WHERE hire_date BETWEEN DATE '2002-01-01' AND DATE '2003-12-31'
+    );
+
 --39. the first and last name, salary, and department ID for all those employees who earn more than the average salary and arrange the list in descending order on salary.
+SELECT employee_id, first_name, last_name, salary, department_id
+FROM employees
+WHERE salary > (
+    SELECT AVG(salary)
+    FROM employees
+)
+ORDER BY salary DESC;
+
 --40. the first and last name, salary, and department ID for those employees who earn more than the maximum salary of a department which ID is 40.
+SELECT employee_id, first_name, last_name, salary, department_id
+FROM employees
+WHERE salary > (
+    SELECT MAX(salary)
+    FROM employees
+    WHERE department_id = 40
+);
+
 --41. the department name and Id for all departments where they located, that Id is equal to the Id for the location where department number 30 is located.
+SELECT department_id, department_name
+FROM departments
+WHERE location_id = (
+    SELECT location_id
+    FROM departments
+    WHERE department_id = 30
+);
+
 --42. the first and last name, salary, and department ID for all those employees who work in that department where the employee works who hold the ID 201.
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE department_id = (
+    SELECT department_id
+    FROM employees
+    WHERE employee_id = 201
+);
+
 --43. the first and last name, salary, and department ID for those employees whose salary is equal to the salary of the employee who works in that department which ID is 40.
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE salary IN (
+    SELECT salary
+    FROM employees
+    WHERE department_id = 40
+);
+
 --44. the first and last name, salary, and department ID for those employees who earn more than the minimum salary of a department which ID is 40.
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE salary > (
+    SELECT MIN(salary)
+    FROM employees
+    WHERE department_id = 40
+);
+
 --45. the first and last name, salary, and department ID for those employees who earn less than the minimum salary of a department which ID is 70.
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE salary > (
+    SELECT MIN(salary)
+    FROM employees
+    WHERE department_id = 40
```

Comment:
The condition requires department 70, not 40.

Reply:
Fixed